playground/src/editor.tsx

```diff
@@ -132,7 +150,14 @@ export function Editor(props: {
     editor.current = newEditor;
     const srcModel = monaco.editor.createModel(props.code, "qsharp");
     newEditor.setModel(srcModel);
-    srcModel.onDidChangeContent(() => checkRef.current());
+    srcModel.onDidChangeContent(() => hirRef.current());
+    srcModel.onDidChangeContent(async () => {
+      await props.languageService.updateDocument(
```

Comment:
This is a little incorrect as this useEffect only runs once on first render (as the dependency array on line 173 is empty), but you closure captures the `props` binding, and thus you'll only every reference the language service that got passed in on first render.

If the language service never changes, this is benign for now. I can fix this up when I add code to restart the workers on any panics. But maybe add a TODO though so I don't forget.

Reply:
What if I just pass in `props.languageService` in the dependency array? Still not pretty, but maybe more correct than a TODO

Reply:
But then you'll recreate the edtior and srcModel if that does change. I think just the comment/note is easiest for now, being that it shouldn't change currently.

Reply:
Oh. I think that's going to have unintended consequences. I'll add my TODO and leave it in your capable hands.